### PR TITLE
[Scheduler] Allow canceling in menu

### LINKED ISSUE #119: [V3 Scheduler] Cancel task from [p]showscheduled menu
Like in title, I think it would be useful to be able to cancel/unschedule task from the menu which shows when you use `[p]showscheduled` or `[p]scheduleradmin viewall`:
Discord_2019-03-17_18-36-23

### PR DESCRIPTION
**Pull request by DiscordLiz**
closes #119

### CODE REVIEW

**Comment by mikeshardmind** [comment]
@DiscordLiz I didn't do a full review of this, just went to load and try on the test instance, but I'm getting a traceback here

```
Exception in command 'showscheduled'
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/discord/ext/commands/core.py", line 79, in wrapped
    ret = await coro(*args, **kwargs)
  File "/root/.local/share/Red-DiscordBot/cogs/CogManager/cogs/scheduler/scheduler.py", line 325, in showscheduled
    await self.task_menu(ctx, tasks)
  File "/root/.local/share/Red-DiscordBot/cogs/CogManager/cogs/scheduler/scheduler.py", line 361, in task_menu
    await menu(ctx, embeds, controls)
  File "/usr/local/lib/python3.7/dist-packages/redbot/core/utils/menus.py", line 64, in menu
    raise RuntimeError("Function must be a coroutine")
RuntimeError: Function must be a coroutine

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/discord/ext/commands/bot.py", line 859, in invoke
    await ctx.command.invoke(ctx)
  File "/usr/local/lib/python3.7/dist-packages/discord/ext/commands/core.py", line 725, in invoke
    await injected(*ctx.args, **ctx.kwargs)
  File "/usr/local/lib/python3.7/dist-packages/discord/ext/commands/core.py", line 88, in wrapped
    raise CommandInvokeError(exc) from exc
discord.ext.commands.errors.CommandInvokeError: Command raised an exception: RuntimeError: Function must be a coroutine
```
If you didn't pass a synchronous function, there may be an upstream bug with the menus.

**Comment by DiscordLiz** [comment]
I passed a coroutine wrapped with functools.partial to get around the fact that the menu doesn't allow passing extra data with kwargs.

I'll open an issue upstream and possibly a PR

**Comment by DiscordLiz** [comment]
Can re-test after Cog-Creators/Red-DiscordBot/issues/2718

File: scheduler/scheduler.py [hunk 1]
```diff
@@ -321,15 +322,43 @@ def pred(item):
...
             t.to_embed(index=i, page_count=count, color=color)
             for i, t in enumerate(tasks, 1)
         ]
 
-        await menu(ctx, embeds, DEFAULT_CONTROLS)
+        controls = DEFAULT_CONTROLS.copy()
+        page_mapping = {i: t for i, t in enumerate(tasks)}
+        actual_task_killer = functools.partial(task_killer, self, page_mapping)
+        controls.update({"\N{NO ENTRY SIGN}": actual_task_killer})
+        await menu(ctx, embeds, controls)
```
Comment (mikeshardmind) [inline scheduler/scheduler.py]:
```suggestion
        await menu(ctx, embeds, controls, message=message)
```

**Comment by mikeshardmind** [comment]
Went ahead and stuck the changes on the test instance (both this and the upstream PR). Will merge at 3.1.2 (pending changes to minimum version)

**Pull request merged**